### PR TITLE
Finished-Week-5-AJG

### PR DESCRIPTION
**Pull request by ajgahan19**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [ ] JavaScript Coding Challenges
- [ ] Ruby Coding Challenges
- [ ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -8,45 +8,171 @@
...
+//     Test Suites: 1 failed, 1 total
+//     Tests:       1 failed, 1 total
+
+const secretCode = (string) => {
+    let four = /a/ig
+    let three = /e/ig
+    let one = /i/ig
+    let zero = /o/ig
+    return string.replaceAll(four, 4).replaceAll(three, 3).replaceAll(one, 1).replaceAll(zero, 0)
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:70]: Nice!

File: code-challenges.test.js [hunk 2]
```diff
@@ -8,45 +8,171 @@
...
+            return true
+            // on the other hand, in order to make another true statement you would need the first two integers and the last three integers to match
+        } else if (array[0] === array[1] && array[2] === array[4]) {
+            return true
+            //if you don't have a matching pair and three of a kind, you would return false
+        } else {
+            return false
+        }
+    }
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:175]: Really similar to how I solved it!

File: ruby-challenges.rb [hunk 3]
```diff
@@ -30,3 +63,32 @@
...
+  def pedal_faster(number)
+    @current_speed = number
+    end
+  def brake(number)
+    if @current_speed > number
+    @current_speed = @current_speed - number
+    elsif @current_speed < number
+      @current_speed = 0
+  end
+  end
```
Comment (WalkerTexas-coder) [inline ruby-challenges.rb:87]: Well done!

File: interview-questions.md [hunk 4]
```diff
@@ -8,44 +8,51 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-4. What are primary keys? Why are they important?
 
-Your answer:
+4. What are primary keys? Why are they important? 
 
-Researched answer:
+Your answer: Primary keys are a notation device. It is used to indicate that it belongs to one string of information. This is so that strings of information that have the same name do not correspond with each other mixing up the data and making it hard to read. Primary keys allow us to differentiate between data even if the information is exactly the same.
+
+Researched answer: They are unique identifiers for each table record. They can be used to quickly look through a table for a piece of data. Only one primary key is assigned to one table. They are important to relational databases as a table may contain many records in which duplicates occur. It is imperative that the primary key is utilized to differentiate the two duplicates from each other.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:35]: Awesome!

**Review by WalkerTexas-coder — APPROVED** [review]
Andre this is an excellent Assessment and I am marking it as Advanced Beginner. Great work- keep it up.

**Comment by ajgahan19** [comment]
Thank you for the comments and all the help this week.

**Pull request merged**